File: ObjectiveC/Gauges101/Gauges101/XGGettingStartedViewController.m
//
//  XGGettingStartedViewController.m
//  Gauges101
//
//  Copyright © 2016 GrapeCity. All rights reserved.
//

#import "XGGettingStartedViewController.h"

@import XuniGaugeDynamicKit;

@interface XGGettingStartedViewController ()
@property (weak, nonatomic) IBOutlet XuniRadialGauge *radialGauge;
@property (weak, nonatomic) IBOutlet XuniBulletGraph *bulletGraph;
@property (weak, nonatomic) IBOutlet XuniLinearGauge *linearGauge;
@property (weak, nonatomic) IBOutlet UIStepper *valueStepper;

@property (weak, nonatomic) IBOutlet UILabel *valueDisplay;

@end

@implementation XGGettingStartedViewController

- (IBAction)stepperValueChanged:(id)sender {
    _valueDisplay.text = [NSString stringWithFormat:@"%i", (int)_valueStepper.value];
    _radialGauge.value = _valueStepper.value;
    _linearGauge.value = _valueStepper.value;
    _bulletGraph.value = _valueStepper.value;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _valueDisplay.text = [NSString stringWithFormat:@"%i", (int)_valueStepper.value];
    
    _radialGauge.loadAnimation.duration = 2;
    _radialGauge.updateAnimation.duration = 0.5;
    
     [_radialGauge.gaugeValueChanged addHandler:^(XuniEventContainer *eventContainer) {
         _valueStepper.value = _radialGauge.value;
         [self stepperValueChanged:nil];
     } forObject:self];
    
    
    
    _linearGauge.loadAnimation.duration = 2;
    _linearGauge.updateAnimation.duration = 0.5;
    
    
    [_linearGauge.gaugeValueChanged addHandler:^(XuniEventContainer *eventContainer)
     {
         _valueStepper.value = _linearGauge.value;
         [self stepperValueChanged:nil];
     }
                                     forObject:self];
    

    _bulletGraph.loadAnimation.duration = 2;
    _bulletGraph.updateAnimation.duration = 0.5;
    
    [_bulletGraph.gaugeValueChanged addHandler:^(XuniEventContainer *eventContainer)
     {
         _valueStepper.value = _bulletGraph.value;
         [self stepperValueChanged:nil];
     }
                                     forObject:self];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Swift/Gauges101/Gauges101/XGGettingStartedViewController.swift
//
//  XGGettingStartedViewController.h
//  Gauges101
//
//  Copyright © 2016 GrapeCity. All rights reserved.
//
import UIKit
import XuniGaugeKit

class XGGettingStartedViewController: UIViewController {

    @IBAction func stepperValueChanged(sender: AnyObject) {
        self.valueDisplay.text = "\(Int(valueStepper.value))"
        self.radialGauge.value = valueStepper.value
        self.linearGauge.value = valueStepper.value
        self.bulletGraph.value = valueStepper.value
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        self.valueDisplay.text = "\(Int(valueStepper.value))"
        self.radialGauge.showText = XuniShowText.None
        self.radialGauge.thickness = 0.6
        self.radialGauge.min = valueStepper.minimumValue
        self.radialGauge.max = valueStepper.maximumValue
        self.radialGauge.value = valueStepper.value
        self.radialGauge.loadAnimation.duration = 2
        self.radialGauge.updateAnimation.duration = 0.5
        self.radialGauge.isReadOnly = false
        self.radialGauge.showText = XuniShowText.None
        radialGauge.gaugeValueChanged.addHandler({(sender: NSObject!, args: XuniEventArgs!) -> Void in
            self.valueStepper.value = self.radialGauge.value
            self.stepperValueChanged(self)
        }, forObject: self)
        self.linearGauge.showText = XuniShowText.None
        self.linearGauge.min = valueStepper.minimumValue
        self.linearGauge.max = valueStepper.maximumValue
        self.linearGauge.value = valueStepper.value
        self.linearGauge.loadAnimation.duration = 2
        self.linearGauge.updateAnimation.duration = 0.5
        self.linearGauge.isReadOnly = false
        
        
        linearGauge.gaugeValueChanged.addHandler({(sender: NSObject!, args: XuniEventArgs!) -> Void in
            self.valueStepper.value = self.linearGauge.value
            self.stepperValueChanged(self)
        }, forObject: self)
        self.bulletGraph.showText = XuniShowText.None
        self.bulletGraph.min = valueStepper.minimumValue
        self.bulletGraph.max = valueStepper.maximumValue
        self.bulletGraph.value = valueStepper.value
        self.bulletGraph.loadAnimation.duration = 2
        self.bulletGraph.updateAnimation.duration = 0.5
        self.bulletGraph.isReadOnly = false
        self.bulletGraph.bad = 45
        self.bulletGraph.good = 80
        self.bulletGraph.target = 90
        bulletGraph.gaugeValueChanged.addHandler({(sender: NSObject!, args: XuniEventArgs!) -> Void in
            self.valueStepper.value = self.bulletGraph.value
            self.stepperValueChanged(self)
        }, forObject: self)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBOutlet weak var radialGauge: XuniRadialGauge!
    @IBOutlet weak var bulletGraph: XuniBulletGraph!
    @IBOutlet weak var linearGauge: XuniLinearGauge!
    @IBOutlet weak var valueStepper: UIStepper!
    @IBOutlet weak var valueDisplay: UILabel!
}
//
//  XGGettingStartedViewController.m
//  Gauges101
//
//  Copyright © 2016 GrapeCity. All rights reserved.
//